__tests__/command_helpers/checkRequirement.ts

```diff
@@ -0,0 +1,161 @@
+
+// Setup checkENV
+jest.mock('../../src/extensions/functions/checkFile')
+const checkFile = require('../../src/extensions/functions/checkFile')
+
+const context = require('gluegun')
+
+const badRule = toPairs({
+  "YARN": [{ "rule": "knope", "binary": "yarn" }] 
+})[0]
+
+let fail
+let stop
+let succeed
+
+describe('checkRequirement', () => {
+  beforeEach(() => {
+    fail = jest.fn()
+    stop = jest.fn()
+    succeed = jest.fn()
+    
+    const spinner = {
+      fail,
+      stop,
+      succeed
+    }
+
+    solidarityExtension(context)
+    context.print = {
+      spin: jest.fn(() => spinner),
+      error: jest.fn()
+    }
+  })
+
+  test('there is a spinner message', async () => {
+    const result = await checkRequirement(badRule, context)
+    expect(context.print.spin.mock.calls).toEqual([
+      ["Verifying YARN"]
+    ])    
+  })
+
+  test('when an invalid rule is given', async () => {
+    const result = await checkRequirement(badRule, context)
+    expect(result).toEqual(['Encountered unknown rule \'knope\''])
+  })
+
+  describe('when rule: cli', () => {
+    beforeEach(() => checkCLI.mockClear());
+    
+    test('sad path', async () => {
+      checkCLI.mockImplementation(async () => "Everything is broken")
+      
+      const rule =  toPairs({ 
+        "YARN": [{ "rule": "cli", "binary": "yarn" }] 
+      })[0]
+      const result = await checkRequirement(rule, context)
+      expect(result).toEqual(["Everything is broken"])
+    })
+
+    test('happy path', async () => {
+      checkCLI.mockImplementation(async () => false)
+      
+      const rule =  toPairs({ 
+        "YARN": [{ "rule": "cli", "binary": "yarn" }] 
+      })[0]
+      const result = await checkRequirement(rule, context)
+      expect(result).toEqual([[]])
+    })
+  })
+
+  describe('when rule: dir', () => {
+    beforeEach(() => checkDir.mockClear());
+    
+    test('happy path', async () => {
+      checkDir.mockImplementation(async () => "It worked!")
+      
+      const rule =  toPairs({ 
+        "YARN": [{ "rule": "dir", "binary": "yarn" }] 
+      })[0]
+      const result = await checkRequirement(rule, context)
+      expect(result).toEqual([[]])
+    })
+
+    test('sad path', async () => {
+      checkDir.mockImplementation(async () => undefined)
+
+      const rule =  toPairs({ 
+        "YARN": [{ "rule": "dir", "binary": "yarn" }] 
+      })[0]
+      const result = await checkRequirement(rule, context)
+      expect(result).toEqual([[]])
+    })
+  })
+
+  describe('when rule: env', () => {
+    beforeEach(() => checkENV.mockClear());    
+
+    test('happy path', async () => {
+      checkENV.mockImplementation(async () => "It worked!")
+      
+      const rule =  toPairs({ 
+        "YARN": [{ "rule": "env", "binary": "yarn" }] 
+      })[0]
+      const result = await checkRequirement(rule, context)
+      expect(result).toEqual([[]])
+    })
+    
+    test('sad path', async () => {
```

Comment:
this made me laugh

Reply:
Yeah, I could probably be more explicit about whats actually happening in these tests. I'll revisit after I reach the coverage goal on this PR.

Reply:
All good. Sad path accurate describes most of my code.

Reply:
Save it for Skellock's code counseling anon meeting.

@mattathompson - looking great!